spec/haskell/src/SEL4/Object/Reply.lhs

```diff
@@ -176,9 +185,8 @@ on the thread state of the replyTCB of the replyPtr
 > replyUnlink_assertion _ _ _ = True
 
 > cleanReply :: PPtr Reply -> Kernel ()
-> cleanReply replyPtr = do
->     reply <- getReply replyPtr
->     setReply replyPtr (reply { replyPrev = Nothing, replyNext = Nothing })
+> cleanReply replyPtr = updateReply replyPtr (\reply -> reply { replyPrev = Nothing,
```

Comment:
I am ok with having `updateReply` at the Haskell level, but I would like to have `cleanReply` to be defined as two separate lines of `updateReply`, one for the prev update and another for the next update. They are quite different semantically and a bit of pain to handle when bundled together...

Reply:
Sounds good!